### DISCUSSION TITLE
Edit by text - Feature

### DISCUSSION THREAD

**Discussion by binaerverkehr** [WyattBlue/auto-editor#590] [2. Feature Ideas]
Dear WyattBlue,

As an enthusiastic user of Auto-Editor, I have a suggestion for a new feature that would extend the tool's functionality: An "Edit By Text" feature. This feature would allow cutting videos based on the transcribed text. For example, one could remove specific words, sentences, or sections from the video by selecting and deleting the corresponding text.

I often produce video tutorials where I speak. It happens regularly that I misspeak, use filler words, or have passages that I would prefer to cut out afterwards. With an "Edit By Text" feature, I could quickly find and remove these parts in the transcript without having to search for them manually. This would significantly speed up my workflow and improve the quality of my videos.

In the CLI tool, parameters like --cut_words "uhm" or --cut_sentence "This was a bad take" could be introduced for this purpose. In the GUI, it would be awesome if one could see the transcript with timestamps and then cut out specific parts of the video with a click or key combination.

This feature would have a great added value for me personally. While there are some providers that offer such a feature, they are often very expensive. An integration into Auto-Editor would allow me and many other content creators to optimize our videos more easily and quickly without having to dig deep into our pockets.

I am using the latest version of Auto-Editor.

Best regards,
Jay

**Comment by binaerverkehr** [comment]
I forgot to mention: i could be implemented by integration of whisperx

**Comment by utopiatopia** [comment]
You can already do this with little monetary cost, although it’s a bit manual. You can use a audio transcription tool (like whispercpp) to transcribe the audio, mux it in with the video, then use
`--edit subtitle:regex=um` or similar.

Related, I don’t believe this is a good idea. This program isn’t a full-fledged video editor, nor is it a transcription app. Implementing these significantly inflate the project’s size and broadens the scope of the project.

**Reply by jmbeach** [reply]
Is the --edit subtitle=regex feature documented? I'm not seeing it
